### ISSUE TITLE
Getting only last event callback on multiple channels

### ISSUE THREAD

**Issue by aluisiora** [gtjoseph/mikronode-ng#10]
I have the following example code of the issue:

``` javascript
var MikroNode = require('mikronode-ng');

var connection = new MikroNode.Connection('10.62.0.21','admin','password',{timeout: 300});

connection.closeOnDone = false;

connection.connect((conn) => {
    var chan1 = conn.openChannel();
    var chan2 = conn.openChannel();
    var chan3 = conn.openChannel();

    chan1.closeOnDone = true;
    chan1.on('done', (data) => {
        data = MikroNode.parseItems(data);
        console.log(data);
    });
    chan1.write(['/ip/address/add','=interface=ether4','=address=192.168.5.5/24']);

    chan2.closeOnDone = true;
    chan2.on('done', (data) => {
        data = MikroNode.parseItems(data);
        console.log(data);
    });
    chan2.write(['/ip/address/add','=interface=ether4','=address=192.168.5.6/24']);

    chan3.closeOnDone = true;
    chan3.on('done', (data) => {
        data = MikroNode.parseItems(data);
        console.log(data);
    });
    chan3.write(['/ip/address/add','=interface=ether4','=address=192.168.5.7/24']);
});
```

But I'm getting data only from 'done' event on the last channel. I tried syncing on one single channel and it works, but apparently not on multiple ones.
I don't know if that's an issue or not, maybe something like this doesn't work?

Thanks in advance =)

**Comment by gtjoseph** [comment]
Hmmmm.  It should work.  I can test it later today.

**Comment by aluisiora** [comment]
So, I've messed around the api a little bit and I think I [pinpointed where the issue occurs](https://github.com/f5eng/mikronode-ng/blob/72340cdaf9fee4a0c5488ae6e856bfd1c9803374/lib/connection.js#L413).

The payload doesn't actually ends until the last command finishes, and then "data.length" returns 0 (or false).
With the example I provided, the connection thinks that all the data that came through is for the chan3, thus ignoring (or overwriting) all the other ".tags=".
